extkingpin/pathorcontent_reloader.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) The EfficientGo Authors.
+// Licensed under the Apache License 2.0.
+
+// Taken from Thanos project.
+//
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package extkingpin
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+)
+
+// logger is an interface compatible with go-kit/logger.
+type logger interface {
+	Log(keyvals ...interface{}) error
+}
+
+// pathOrContent is an interface compatible with PathOrContent.
+type pathOrContent interface {
+	Content() ([]byte, error)
+	Path() string
+}
+
+// PathContentReloader starts a file watcher that monitors the file indicated by pathOrContent.Path() and runs
+// reloadFunc whenever a change is detected.
+// A debounce timer can be configured via function args to handle situations where many events that would trigger
+// a reload are receive in a short period of time. Files will be effectively reloaded at the latest after 2 times
+// the debounce timer. By default the debouncer timer is 1 second.
+// To ensure renames and deletes are properly handled, the file watcher is put at the file's parent folder. See
+// https://github.com/fsnotify/fsnotify/issues/214 for more details.
+func PathContentReloader(ctx context.Context, fileContent pathOrContent, debugLogger logger, errorLogger logger, reloadFunc func(), debounceTime time.Duration) error {
```

Comment:
```suggestion
func StartPathContentReloader(ctx context.Context, fileContent pathOrContent, debugLogger logger, errorLogger logger, reloadFunc func(), debounceTime time.Duration) error {
```

I think we need verb to tell what's going on 🤔